# Introduction to visualising Next Generation Sequencing data with IGV

# Description 

Modern genomics technologies are able to produce large volumes of data that often leave researchers feeling overwhelmed and unsure of how to begin the process of biological interpretation.

In this course, we explain the common file formats generated by sequencing technologies and how they can be manipulated and explored by non bioinformaticians. The tools that we will use is the Integrative Genomics Viewer (IGV) (http://software.broadinstitute.org/software/igv/).

If time allows, there will be time at the end of the session for you to explore your own datasets with the assistance of the instructors.

This event is part of a series of training courses organized in collaboration with Dr. Mark Dunning at CRUK Cambridge Institute. The materials for the course were developed in collaboration with Dr. Thomas Carroll from the MRC CSC. 

# Objectives: After the course you should be able to:-

- Recognise and describe the contents of common file formats used for NGS data
- Use IGV to explore your own sequencing data
- Navigate to, and explore, regions of interest in a reference genome

# Aims: During the course you will learn about

- Common File formats used to store NGS data
- How to run IGV
- Navigating the IGV interface and moving around genomes
- Loading and visualising data in IGV
- Exploring RNA-seq and ChIP-seq data in IGV
- Importing and interpreting Variant calls in IGV
